### PR TITLE
make the facebook gating prod-only

### PR DESCRIPTION
**Pull request by oconnor663**
In gating only the proof creation codepaths, I forgot to restrict those gates to prod mode.

r? @mlsteele @maxtaco

### CODE REVIEW

File: go/externals/services.go [hunk 1]
```diff
@@ -23,11 +23,11 @@ func (e externalServicesCollection) GetServiceType(s string) libkb.ServiceType {
 	return e[strings.ToLower(s)]
 }
 
-func (e externalServicesCollection) ListProofCheckers() []string {
+func (e externalServicesCollection) ListProofCheckers(g *libkb.GlobalContext) []string {
```
Comment (maxtaco) [inline go/externals/services.go]: just pass run mode, not G

File: go/client/cmd_prove.go [hunk 2]
```diff
@@ -104,7 +104,7 @@ func (p *CmdProve) installOutputHook(ui *ProveUI) {
 
 // NewCmdProve makes a new prove command from the given CLI parameters.
 func NewCmdProve(cl *libcmdline.CommandLine, g *libkb.GlobalContext) cli.Command {
-	serviceList := strings.Join(g.Services.ListProofCheckers(), ", ")
+	serviceList := strings.Join(g.Services.ListProofCheckers(g.GetRunMode()), ", ")
```
Comment (oconnor663) [inline go/client/cmd_prove.go]: This call to `g.GetRunMode()` doesn't seem to actually respect the `--run-mode` flag. I need to dig into it.

**Comment by maxtaco** [comment]
I can't imagine that is true. Have you restarted you daemon?

**Comment by oconnor663** [comment]
Ah ok, this is happening because help strings are computed very early in main, before we parse command line flags. Not a big deal. Almost no one in the wild will be using `--run-mode`, and even then all it does is change the help text.

**Pull request merged**